### PR TITLE
Add Featured pages

### LINKED ISSUE #65: Build featured mentors page for user
child of #53

### LINKED ISSUE #66: Build featured users page for mentor
child of #53

### PR DESCRIPTION
**Pull request by rmjordas**
**TODO**:

- [x] ~~Determine~~ Fix path for `MentorFeatured` and `UserFeatured`
- [x] Fix `MentorFeatured` and `UserFeatured` to use bottom navigation component
- [x] Replace hard-coded featured user/mentor lists
- [x] Remove `<Spacer />`
- [x] Update mock API response for featured users and mentors
- [x] Pull method for generating ratings to its own `<Rating />` component
- [x] Fix styling issues

Closes #65 
Closes #66

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -26,10 +28,12 @@ class App extends React.Component {
           <Route exact={true} path="/auth/linkedin/login" component={MentorLogin} />
           <Route exact={true} path="/mentor" component={JoinMentor} />
           <Route exact={true} path="/mentor/edit" component={MentorEdit} />
+          <Route exact={true} path="/user/featured" component={UserFeatured} />
```
Comment (roymoran) [inline src/components/App.tsx]: For now the UserFeatured component should be placed in the user.tsx file 

changed this line `<Route path={`/mentor/:id/${DASHBOARD_VALUE}`} component={MentorRequestsList} />`
to `<Route path={`/mentor/:id/${DASHBOARD_VALUE}`} component={UserFeatured} />`

File: src/components/App.tsx [hunk 2]
```diff
@@ -26,10 +28,12 @@ class App extends React.Component {
...
           <Route exact={true} path="/user/edit" component={UserEdit} />
           <Route exact={true} path="/user/signup" component={UserSignup} />
           <Route exact={true} path="/user/login" component={UserLogin} />
           <Route exact={true} path="/user" component={FindMentor} />
+          <Route exact={true} path="/mentor/featured" component={MentorFeatured} />
```
Comment (roymoran) [inline src/components/App.tsx]: For now the MentorFeatured component should be placed in the mentor.tsx file

changed this line <Route path={/mentor/:id/${DASHBOARD_VALUE}} component={MentorRequestsList} />
to <Route path={/mentor/:id/${DASHBOARD_VALUE}} component={MentorFeatured} />

File: src/components/App.tsx (hunk 1 above)
Comment (roymoran) [reply src/components/App.tsx]: If no other file is using the UserRequestsList component then we should delete it.

File: src/components/App.tsx (hunk 2 above)
Comment (roymoran) [reply src/components/App.tsx]: Same here if no other file is using the MentorRequestsList we should delete it.

File: src/components/mentor/mentorFeatured.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        }
+      ]
+    };
+  }
+
+  public render() {
+    return (
+      <React.Fragment>
+        <MainPageTitle title="Featured" />
+        <Spacer height={42} />
```
Comment (roymoran) [inline src/components/mentor/mentorFeatured.tsx]: I tested the page without the Spacer component. I think it looks better. Let me know what you think. If it looks better we can remove it.

File: src/components/user/userFeatured.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        }
+      ]
+    };
+  }
+
+  public render() {
+    return (
+      <React.Fragment>
+        <MainPageTitle title="Featured" icon="search" path="#" />
+        <Spacer height={42} />
```
Comment (roymoran) [inline src/components/user/userFeatured.tsx]: same here. Let's consider removing the Spacer component.

File: src/components/user/userFeatured.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+const Spacer = styled.div`
+  height: ${(p: { height: number }) => `${p.height}px`};
+`;
+
+export class UserFeatured extends React.Component<TRouteProps, IUserFeaturedState> {
+  public constructor(props: TRouteProps) {
+    super(props);
+
+    this.state = {
```
Comment (roymoran) [inline src/components/user/userFeatured.tsx]: Let's mock out a single api call that will return the list of featured mentors. And for now we will just return a mocked response until I build out the api endpoint for this. You'll have to update the api interface, and both apiService.ts and mockApiService.ts.

**Comment by roymoran** [comment]
@ruj here is where you will put the route for each component 

MentorFeatured will go in mentor.tsx line 111 -> <Route path={`/user/:id/${DASHBOARD_VALUE}`} component={MentorFeatured} />

UserFeatured will go in user.tsx line 105 -> <Route path={`/user/:id/${DASHBOARD_VALUE}`} component={UserFeatured} />

File: src/components/mentor/mentorFeatured.tsx (hunk 3 above)
Comment (rmjordas) [reply src/components/mentor/mentorFeatured.tsx]: Yeah, I just copied what's in the mock up. I also think it looks better without `<Spacer />`.

File: src/components/shared/featured.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        <FeaturedContent>
+          <FeaturedName>{this.props.name}</FeaturedName>
+          <FeaturedTitle>{this.props.title}</FeaturedTitle>
+          {this.generateRating()}
+        </FeaturedContent>
+      </StyledFeatured>
+    );
+  }
+
+  private generateRating = () => {
```
[the remark this answers is not in the document]
Comment (rmjordas) [reply src/components/shared/featured.tsx]: The `<Featured />` component is provided props with a shape defined below:

https://github.com/projectunic0rn/mentorship-ui/blob/947d0646e2892cfbe4cadd0e0b6495c560c3ede8/src/components/shared/featured.tsx#L7-L12 

The user and mentor ratings are provided in the state. The rating can be changed there.

https://github.com/projectunic0rn/mentorship-ui/blob/947d0646e2892cfbe4cadd0e0b6495c560c3ede8/src/components/user/userFeatured.tsx#L26-L42

https://github.com/projectunic0rn/mentorship-ui/blob/947d0646e2892cfbe4cadd0e0b6495c560c3ede8/src/components/mentor/mentorFeatured.tsx#L26-L42

[the remark this answers is not in the document]
Comment (roymoran) [reply src/components/shared/featured.tsx]: got it. thanks i missed that part of the code.

**Pull request merged**